src/main/java/Duke/Parser.java

```diff
@@ -0,0 +1,167 @@
+    private final static String COMMAND_ADD_DEADLINE = "deadline";
+    private final static String COMMAND_ADD_EVENT = "event";
+    private final static String COMMAND_DELETE_EVENT = "delete";
+    private final static String COMMAND_FIND = "find";
+    private final static String DATE_DEADLINE = "/by";
+    private final static String DATE_EVENT = "/at";
+
+    private final static DateTimeFormatter SAVE_READ_DATETIME_FORMAT = DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm");
+
+    /**
+     * Creates a Parser object.
+     */
+    public Parser() {
+    }
+
+    /**
+     * Types of task that could result from the parsed user command.
+     */
+    enum TypeOfCommand {
+        DONE,
+        DEADLINE,
+        TODO,
+        EVENT,
+        DELETE,
+        FIND
+    }
+
+    /**
+     * Parses the input from the user and returns a command with respect to the
+     * input from the user.
+     *
+     * @param fullCommand the given input from the user.
+     * @return the command that will be carried out by Duke.
+     */
+    public static Command parse(String fullCommand) {
+        String shortCommand;
+        String restOfCommand;
+        if (fullCommand.contains(" ")) {
+            shortCommand = fullCommand.substring(0, fullCommand.indexOf(' '));
+            restOfCommand = fullCommand.substring(shortCommand.length() + 1);
+        } else {
+            shortCommand = fullCommand;
+            restOfCommand = "";
+        }
+        switch (shortCommand) {
+        case COMMAND_LIST:
+            return new ListCommand();
+        case COMMAND_TERMINATE:
+            if (restOfCommand.isEmpty()) {
+                return new ExitCommand();
+            } else {
+                return new IncorrectCommand("OOPS !!! Lo siento, pero no sé qué significa eso :-(");
+            }
+        case COMMAND_COMPLETE_TASK:
+            return handle(restOfCommand, TypeOfCommand.DONE);
+        case COMMAND_ADD_DEADLINE:
+            return handle(restOfCommand, TypeOfCommand.DEADLINE);
+        case COMMAND_ADD_TODO:
+            return handle(restOfCommand, TypeOfCommand.TODO);
+        case COMMAND_ADD_EVENT:
+            return handle(restOfCommand, TypeOfCommand.EVENT);
+        case COMMAND_DELETE_EVENT:
+            return handle(restOfCommand, TypeOfCommand.DELETE);
+        case COMMAND_FIND:
+            return handle(restOfCommand, TypeOfCommand.FIND);
+        default:
+            return new IncorrectCommand("OOPS !!! Lo siento, pero no sé qué significa eso :-(");
+        }
+    }
+
+    /**
+     * Handles the creation of more complicated commands that require interaction with the TaskList.
+     *
+     * @param restOfCommand the content of the task to be added after initial command.
+     * @param TypeOfCommand the type of command to be carried out.
+     * @return the command that will finally be carried out by Duke.
+     */
+    private static Command handle(String restOfCommand, TypeOfCommand TypeOfCommand) {
```

Comment:
Should handle be further split up into separate methods? I believe the handle method was created to help shorten the parse method, however that resulted in handle being a long method.

Reply:
Ok, I will consider that! Thanks!